certbot/certbot/_internal/cert_manager.py

```diff
@@ -221,9 +221,14 @@ def cert_path_to_lineage(cli_config):
     :raises `errors.Error`: If the specified cert path can't be matched to a lineage name.
     :raises `errors.OverlappingMatchFound`: If the matched lineage's archive is shared.
     """
+    # cli_config.cert_path might be a (path, contents) tuple or just a path.
+    cert_path = cli_config.cert_path[0] \
+                if isinstance(cli_config.cert_path, tuple) \
```

Comment:
our meeting notes say "make sure if i pass an empty tuple, this is covered by an integration test", which I'm assuming refers to this part. I don't remember how this could be an empty tuple, but you might?

Reply:
Mmmm. When I was playing with #8671, I was pondering removing `revoke`'s special treatment of `{cert,key}_path` in `_paths_parser`, which dates back to 2015. Doing away with all this tuple business.

If it's going to make both changes easier, maybe it's worth doing. WDYT?

Reply:
Ah, right. I'd be down for that, if we can make sure that there's not actually a good reason for it, and it's not part of our public api. Otherwise we should just add a test here and be done with it, and maybe a type hint on that value somewhere so we know the two things it might be, since it's such a weird and unexpected variation.

Reply:
The tuple made its farewells in #8687 so I think this part is resolved.